From: Ben Hutchings <benh@debian.org>
Date: Mon, 25 Nov 2024 22:02:01 +0100
Subject: Revert "copy-firmware.sh: remove no longer reachable test -f"
Fowarded: not-needed

This reverts commit 25d5e6352ba5dd8d8d75cb80fc905f65b13c4c44.  For any
files that we exclude, copy-firmware.sh will now create empty files
under debian/build/install.  Restore the check for missing files.

An alternative to this would be to patch the excluded files out of
WHENCE, but this would be a far larger patch that would probably
require more work to maintain.

---
--- a/copy-firmware.sh
+++ b/copy-firmware.sh
@@ -72,6 +72,7 @@ fi
 
 # shellcheck disable=SC2162 # file/folder name can include escaped symbols
 grep -E '^(RawFile|File):' WHENCE | sed -E -e 's/^(RawFile|File): */\1 /;s/"//g' | while read k f; do
+    test -f "$f" || continue
     install -d "$destdir/$(dirname "$f")"
     $verbose "copying/compressing file $f$compext"
     if test "$compress" != "cat" && test "$k" = "RawFile"; then
